### DISCUSSION TITLE
Android 14: Get modal "This app was built for an older version of Android"

### DISCUSSION THREAD

**Discussion by corneliusroemer** [NightscoutFoundation/xDrip#3135] [Bug reports] [answered]
I get a warning when starting on Android 14:
>×Drip+
This app was built for an older version of
Android. It might not work properly and
doesn't include the latest security and privacy
protections. Check for an update, or contact the
app's developer.

![Screenshot_20231017-161504](https://github.com/NightscoutFoundation/xDrip/assets/25161793/96d65476-b985-437c-857a-fbbb10ef10f8)

**Comment by Navid200** [comment, marked as answer]
Yes, we know that:  https://github.com/NightscoutFoundation/xDrip/discussions/3114
And it will be addressed.

You need to approve it.  And after you do that, you will not get that warning again.
Will you?

**Reply by corneliusroemer** [reply]
Yep, only got it once

**Reply by corneliusroemer** [reply]
Is this "Not made for Android14" modal related to not being able to install the watch component with Andoid 14 and WearOS 4?

**Reply by DevanWolf** [reply]
The targetSdkVersion in build.gradle is set to 24 instead of 34

**Reply by Navid200** [reply]
Yes, that's correct, and that's why xDrip can be used on Android 7 and newer.
It used to be that you could also use it on Android 6.  We had to change that so that Android 15 can be used, which is expected to be released shortly.  

We always move xDrip forward while resisting Google's push to force everyone to move to newer versions of Android.
That would force people to buy new phones.  xDrip is used worldwide.  Some may not afford to buy a new phone frequently.
Sooner or later, we will have to retire Android 7 as well.  But, my hope is to delay that as much as possible.

Anyway, you have two options to address that notification.  One, my recommendation, is explained here:
https://navid200.github.io/xDrip/docs/FAQ/UnsafeAppBlocked.html  

The other is to disable Google Play protect.
